src/main/java/com/rapid7/client/dcerpc/mslsad/LocalSecurityAuthorityService.java

```diff
@@ -117,10 +119,38 @@ public void closePolicyHandle(final byte[] handle) throws IOException {
         callExpect(request, "LsarClose", SystemErrorCode.ERROR_SUCCESS, SystemErrorCode.STATUS_INVALID_HANDLE);
     }
 
-    public LsarLookupNamesResponse lookupNames(final byte[] policyHandle, final String... names)
+    /**
+     *
+     * @param policyHandle Handle to the policy
+     * @param names Names to lookup SIDs for
+     * @return Returns a list of SIDs
+     * @throws IOException Thrown if exception happens at the RPC layer
+     */
+    public SID[] lookupNames(final byte[] policyHandle, final String... names)
             throws IOException {
         final LsarLookupNamesRequest request = new LsarLookupNamesRequest(parseHandle(policyHandle), names);
-        return callExpectSuccess(request, "LsarLookupNames");
+        LsarLookupNamesResponse response = callExpect(request, "LsarLookupNames",
+                SystemErrorCode.ERROR_SUCCESS,
+                SystemErrorCode.STATUS_SOME_NOT_MAPPED);
+        LSAPRTranslatedSID[] translatedSIDs = response.getLsaprTranslatedSIDs().getLsaprTranslatedSIDArray();
+        LSAPRTrustInformation[] domainArray = response.getLsaprReferencedDomainList().getLsaprTrustInformations();
+        // Create DTO SIDs
+        SID[] sids = new SID[translatedSIDs.length];
+        for (int i = 0; i < translatedSIDs.length; i++) {
+            try {
+                if (translatedSIDs[i] != null) {
```

Comment:
Like @bli-r7 mentioned, I think we'll want to insert null here in this case too. As otherwise the name entries won't line up with the sids.
```java
LSAPRTranslatedSID translatedSID = translatedSIDs[i];
if (translatedSID != null) {
   //get domain SID
   int domainIndex = (int) translatedSIDs[i].getDomainIndex();
   if (domainIndex >= 0) {
      RPCSID sid = domainArray[domainIndex].getSid();
      SID dtoSID = parseRPCSID(sid);
      //add RID to SID
      sids[i] = dtoSID.resolveRelativeID(translatedSIDs[i].getRelativeId());
      continue;
   }
}
sids[i] = null;
```